ada_feeding_perception/ada_feeding_perception/food_on_fork_categorical_naive_bayes_training.py

```diff
@@ -0,0 +1,288 @@
+import zipfile
+import tempfile
+import shutil
+import os
+import cv2 as cv
+import matplotlib.pyplot as plt
+import numpy as np
+from sklearn.metrics import accuracy_score, confusion_matrix
+from sklearn.naive_bayes import CategoricalNB
+import joblib
+from matplotlib.patches import Rectangle
+import argparse
+
+
+def unzip_file(zip_path, extract_dir):
+    with zipfile.ZipFile(zip_path, "r") as zip_ref:
+        zip_ref.extractall(extract_dir)
+
+
+def get_labels_folders(extracted_folders):
+    train_test_data_folder_path = extracted_folders[0]  # train_test_data
+    train_test_data_folder = os.listdir(
+        os.path.join(temp_dir, train_test_data_folder_path)
+    )
+
+    train_folder_path = train_test_data_folder[0]  # train
+    train_folder = os.listdir(
+        os.path.join(temp_dir, train_test_data_folder_path, train_folder_path)
+    )
+
+    test_folder_path = train_test_data_folder[1]  # test
+    test_folder = os.listdir(
+        os.path.join(temp_dir, train_test_data_folder_path, test_folder_path)
+    )
+
+    train_labels_folder = {}
+    test_labels_folder = {}
+
+    train_folder_path = os.path.join(
+        temp_dir, train_test_data_folder_path, train_folder_path
+    )
+    test_folder_path = os.path.join(
+        temp_dir, train_test_data_folder_path, test_folder_path
+    )
+    for labels_path in train_folder:
+        train_labels_folder[labels_path] = os.listdir(
+            os.path.join(train_folder_path, labels_path)
+        )
+
+    for labels_path in test_folder:
+        test_labels_folder[labels_path] = os.listdir(
+            os.path.join(test_folder_path, labels_path)
+        )
+
+    return train_labels_folder, test_labels_folder, train_folder_path, test_folder_path
+
+
+def normalize_to_uint8(img):
+    # Normalize the image to 0-255
+    img_normalized = ((img - img.min()) / (img.max() - img.min()) * 255).astype("uint8")
+    return img_normalized
+
+
+def plt_show_depth_img(img, show=True, title=""):
+    # Plot the depth image
+    fig, ax = plt.subplots()
+    for i in range(0, 128, 8):
+        for j in range(0, 84, 4):
+            ax.add_patch(Rectangle((i, j), 8, 4, edgecolor="r", linewidth=1, fill=None))
+    plt.imshow(img)
+
+    plt.title(title)
+    plt.colorbar()
+    if show:
+        plt.show()
+
+
+if "__main__" == __name__:
```

Comment:
It's not good coding practice to have all this code in one block. Typically you should separate it into different helper functions.

I'm fine with merging it in as is because this is a script for training so won't be called often. But in general you should avoid these huge blocks of code.

Reply:
I will keep this in mind for the future.

Reply:
will keep in mind for future